ProjFS.Mac/PrjFSKext/PrjFSKext/VirtualizationRoots.cpp

```diff
@@ -56,18 +119,17 @@ kern_return_t VirtualizationRoots_Cleanup()
     return KERN_FAILURE;
 }
 
-VirtualizationRoot* VirtualizationRoots_FindForVnode(vnode_t vnode)
+VirtualizationRootHandle VirtualizationRoot_FindForVnode(vnode_t vnode)
 {
-    VirtualizationRoot* root = nullptr;
+    VirtualizationRootHandle rootIndex = RootHandle_None;
     
     vnode_get(vnode);
     // Search up the tree until we hit a known virtualization root or THE root of the file system
-    while (nullptr == root && NULLVP != vnode && !vnode_isvroot(vnode))
+    while (rootIndex == RootHandle_None && NULLVP != vnode && !vnode_isvroot(vnode))
     {
-        int16_t rootIndex = VirtualizationRoots_LookupVnode(vnode, nullptr);
-        if (rootIndex >= 0)
+        rootIndex = VirtualizationRoots_FindRootAtVnode(vnode, nullptr);
+        if (rootIndex != RootHandle_None)
```

Comment:
Would it be safer to check `!VirtualizationRoot_IsValidRootHandle(rootIndex)`?

(Or is part of the contract with `VirtualizationRoots_FindRootAtVnode` that it will only return `RootHandle_None` or a valid index?)

Reply:
`VirtualizationRoots_FindRootAtVnode()` (via `FindRootAtVnode_Locked()`) [will eventually be able to return `RootHandle_ProviderTemporaryDirectory`](https://github.com/Microsoft/VFSForGit/pull/225/files#diff-86d4504941139fd962dc83a58c348753R205), in which case we've actually found what we're looking for and want to stop the search and return that special handle from this function. This is evidently not obvious, so I'll put in a comment to that effect.